### PR TITLE
add the feature to download `txt` file in Radis App interface

### LINKED ISSUE #637: Download as txt
Tried the download. We'll need a few changes to make it perfectly user friendly : 

.1 Keep the current : `Download .SPEC`, but add info on what to do with the file. 
2. add an alternative button next to it,  `.TXT`   to download as .txt directly 

1.  Download button should with a message to explain what it is and how to read it. Ex  

> ".spec is a compressed JSON format, and the recommended format to work with spectra in Python. Load it locally or in [Radis-Lab](https://radis.github.io/radis-lab/) with `s=radis.load_spec()` then print all calculation conditions (`print(s)`) , rescale or edit the spectrum, compute other spectral arrays, plot in arbitrary units, etc. See Radis Python documentation to know more. 

Implementation : 
- can be a tooltip, since we can have them #184  
- or can be a message shown to the user when clicking download the first time . I like the tooltip option better

2. For the second button, use the `Spectrum.savetxt()` method

### PR DESCRIPTION
**Pull request by arunavabasu-03**
Fixes #637

### CODE REVIEW

**Comment by erwanp** [comment]
Screenshot ? 
Show an extract of the downloaded file, too

**Comment by arunavabasu-03** [comment]
[hitran_CO_1900_2300cm-1_300K_1.01325atm.csv](https://github.com/suzil/radis-app/files/9440086/hitran_CO_1900_2300cm-1_300K_1.01325atm.csv)

**Review by erwanp — APPROVED** [review]

File: frontend/src/components/Form.tsx [hunk 1]
```diff
@@ -186,6 +187,37 @@ export const Form: React.FunctionComponent<FormProps> = ({
...
+        link.click();
+        setDisableDownloadButton(false);
+        const response = await rawResponse.data;
+        if (response.error) {
+          handleBadResponse(response.error);
+        } else {
+          setDisableDownloadButton(false);
+        }
+        setDisableDownloadButton(false);
+        setProgress(100);
```
Comment (suzil) [inline frontend/src/components/Form.tsx:219]: There's too much duplication between this and download-spectrum. Could we find a way to refactor it to have more code re-use?

**Comment by erwanp** [comment]
See https://github.com/suzil/radis-app/issues/637?notification_referrer_id=NT_kwDOAPV-p7M0MjQ0NDA1MjI1OjE2MDg4NzQz#issuecomment-1231155004

**Comment by arunavabasu-03** [comment]
> See #637

can i create another pr to do this 😀 .

**Comment by erwanp** [comment]
sure

File: frontend/src/components/Form.tsx [hunk 2]
```diff
@@ -155,25 +155,42 @@ export const Form: React.FunctionComponent<FormProps> = ({
...
+            "download",
+            `${data.database}_${molecules}_${data.min_wavenumber_range}_${data.max_wavenumber_range}cm-1_${data.tgas}K_${data.pressure}atm.spec`
+          );
+        }
+        if (endpoint === "download-txt") {
+          link.setAttribute(
+            "download",
+            `${data.database}_${molecules}_${data.min_wavenumber_range}_${data.max_wavenumber_range}cm-1_${data.tgas}K_${data.pressure}atm.csv`
+          );
+        }
```
Comment (suzil) [inline frontend/src/components/Form.tsx:192]: looks much better! a lot less duplication of the logic

**Review by suzil — APPROVED** [review]

**Pull request merged**